packages/loader/driver-definitions/src/storage.ts

```diff
@@ -49,8 +49,20 @@ export interface IDeltaStorageService {
     get(
         tenantId: string,
         id: string,
-        from: number,
-        to: number): Promise<IDeltasFetchResult>;
+        from: number, // inclusive
+        to: number // exclusive
+    ): Promise<IDeltasFetchResult>;
+}
+
+/**
+ * Read interface for the Queue
+ */
+ export interface IStream<T> {
+    pop(): Promise<T | undefined>;
+    /**
+     * aborts processing and releases resources
+     */
+    cancel(): void;
```

Comment:
Overall i like the approach.

Some Thoughts:
- Do we need cancel if we take abort signal in readMessage? can't that be used to abort the stream?
- Related, what happens on fatal error? will pop throw? can i retry pop? should the stream be disposable?
- How do i know i've reached the end of the stream? Do i need to count? what if i requested more than exists in storage?
- Related, how long is a stream valid? Can i create a stream to Number.MAX and just keep using it whenever i want ops?

Reply:
All good questions, but we should separate interface (IStread) vs. particular contract (fetching ops).
Some answers:
- cancel() was late addition, I'm not sure I like it, planning to iterate on it. dispose() is likely better name for it. It's different from abort signal passed by caller, as consumer & producer of IStream can be two different entities. That said, there is a case to be made that in all cases we can compose signals and use just signals.
- Queue object that implements IStream has capability to report errors (pop throws) and yes readMessages impl uses it
- undefined is returned on EOF. I can make it more explicit.
- If you ask more than what exists in storage - that's defined not on IStream level, but on particular user level (readMessages). Same behavior as before - you get only what storage knows about.
- RE last question: same as above - it's defined by higher level contract. In case of readMessages, it will keep streaming all the ops. Currently, the way fetching is done is it will keep fetching all the ops. If processing ops takes forever, then yes, we can accumulate million ops. We can put some breaks in there - that would probably be a good thing to have for long run (I'll follow up)

Reply:
Some follow ups:
- removed cancel()
- RE OEF: Changed return result to be same as iterator.next() shape. Ideally we use async iterator here, but I'm hesitant to switch due to potential impact on compat matrix (I definitely got into trouble by using async generators in shipping code in the past).

Reply:
I've also opened https://github.com/microsoft/FluidFramework/issues/5733 to follow up on throttling. This is not a concern today (and not new in this PR), but something I need to address before enabling concurrency > 1 (currently enabled only in fetch-tool).